packages/core/test/unit/Site.test.js

```diff
@@ -389,7 +393,130 @@ describe('Site deploy with Travis', () => {
     const site = new Site('./', '_site');
     await expect(site.deploy(true))
       .rejects
-      .toThrow(new Error('-t/--travis expects a GitHub repository.\n'
+      .toThrow(new Error('-c/--ci expects a GitHub repository.\n'
+        + `The specified repository ${invalidRepoConfig.deploy.repo} is not valid.`));
+  });
+});
+
+describe('Site deploy with AppVeyor', () => {
```

Comment:
Just one suggestion to reduce the test repitition: Since there are only two environment variables which change
1. ci environment identifier (`appveyor/travis`)
2. repo slug environment variable

maybe we could re organize the tests into a [matrix](https://jestjs.io/docs/en/api#testeachtablename-fn-timeout)? so no need to repeat the same test names / structure / etc. for other environments too (gh actions, ...).

Reply:
Thanks for the suggestion @ang-zeyu ! Yes, for these tests the only thing that changes are the environmental variables. I will try this out.

Reply:
@ang-zeyu , I have tried out reorganizing the tests into a matrix, but I feel it may be too restrictive because
1. Cannot write descriptive test names so it may be hard to express in readable terms the motive of the test
2. Some of the tests are slightly different and trying to implement them using the matrix may lead to extra complexity (alot of if/else).
3. Not sure if extra configurations are still needed other CI platforms (gh-actions) so it may not be very extensible.

What do you think?

Reply:
>Cannot write descriptive test names so it may be hard to express in readable terms the motive of the test

Hmm which ones are these? The test names currently don't seem to make references to the environment, save for one. (right below)

> `Site deploy -c/--ci should not deploy if not in Travis`

We can forgo the matrix for this test completely (since all environment variables are being deleted before each test already)

>Some of the tests are slightly different and trying to implement them using the matrix may lead to extra complexity (alot of if/else).

Hmm, could you PR these? I gave it a second look but couldn't find any significant differences needing branches in the travis / appveyor tests. Let me know what I missed out! :-) 

>Not sure if extra configurations are still needed other CI platforms (gh-actions) so it may not be very extensible.

In my opinion its highly unlikely - I don't think there's many ci environments out there that don't support an easy identifier + owner/repo slug environment variable pair / don't support setting environment variables (`GITHUB_TOKEN`).

It's a valid point, but I think we should try to avoid repeating 100+ lines * n environments of tests over this. We can deal with the exceptions as needed separately.

Reply:
@ang-zeyu I have updated the tests to use the matrix. Could you check and see if I have correctly implemented it as what you described?